Add exchange list

michalsmiarowski opened the pull request:
Closes: #335 

This PR adds whole new Exchange List section for the TBTC Website.

**Main changes:**

- Add exchange list section

- Modify `formatRows` function so it can be used by both ExchangeList and Integrations. Rename it `formatRows`-->`formatImageLinkRows` and move it to `utils.js` file

- Add proper styles to make sure all the section is responsive

- change `<div>` tag to `<ul><li>` tag inside `formatImageLinkRows` function so both Exchange List and Integrations will be rendered as unordered list

- Add flexibility for future dark mode mentioned in #338

- Add exchange list to Gatsby so it can be added, edited etc. by CMS

- Add exchange list section title to translation files (will need to be translated - right know it's in english in all languages)

**Notes:**

- for svg files to work properly i had to use [react-svg](https://www.npmjs.com/package/react-svg). It was needed to render the svg file as raw `<svg>` tag so they could be colored in in scss. We need different colors not only for dark mode, but also for hovering the buttons and matcha logo itself has 3 different colors (black, white, gray) that we have to apply.

- the flexibility for dark mode was written based on a #338 Pull Request. It will require similar changes to inverse a colors in darkMode for this section (changing `$black`/`$white` to `var(--body-foreground-color)`/`var(--body-background-color)`) + additionally swapping `--svg-main-color` i `--svg-secondary-color` variables when in dark Mode (same as swapping foreground and background color in #338).

- to test the dark mode look for the exchange list section right now, you have to do 2 things:
    1. Swap values of `$black` and `$white` variables in `src/css/color.scss`
    2. Swap values of `--svg-main-color` and `--svg-secondary-color` in `src/css/typography.scss`

**Examples:**
<details>
  <summary>Basic example</summary>

![image](https://user-images.githubusercontent.com/40306834/101768785-55204780-3ae6-11eb-9cb9-0dfb4986f8eb.png)

On button hover:

![image](https://user-images.githubusercontent.com/40306834/101768811-61a4a000-3ae6-11eb-935f-413b9054119b.png)

</details>
<details>
  <summary>Responsiveness for 4 items</summary>

![image](https://user-images.githubusercontent.com/40306834/101765426-93673800-3ae1-11eb-9dfb-e8248d943346.png)

![image](https://user-images.githubusercontent.com/40306834/101765463-a24dea80-3ae1-11eb-94bc-0717a0a67402.png)

![image](https://user-images.githubusercontent.com/40306834/101765500-ad087f80-3ae1-11eb-8fdd-80b7b28b6570.png)

</details>

<details>
  <summary>Responsiveness for 5 items</summary>

![image](https://user-images.githubusercontent.com/40306834/101767329-5781a200-3ae4-11eb-912d-3a65ad9db7fe.png)

![image](https://user-images.githubusercontent.com/40306834/101767359-62d4cd80-3ae4-11eb-92f3-71fda454774b.png)

![image](https://user-images.githubusercontent.com/40306834/101767375-6b2d0880-3ae4-11eb-8493-e00fe1e12b0b.png)

</details>

<details>
  <summary>Responsiveness for 6 items</summary>

![image](https://user-images.githubusercontent.com/40306834/101767914-32d9fa00-3ae5-11eb-8794-9c7f4a0556a7.png)

![image](https://user-images.githubusercontent.com/40306834/101767935-3bcacb80-3ae5-11eb-8c68-f006cece6689.png)

![image](https://user-images.githubusercontent.com/40306834/101767961-44230680-3ae5-11eb-8881-49b40d62b506.png)

</details>

<details>
  <summary>Responsiveness for 7 items</summary>

![image](https://user-images.githubusercontent.com/40306834/101768194-8d735600-3ae5-11eb-91c8-df295497bfa7.png)

![image](https://user-images.githubusercontent.com/40306834/101768220-96642780-3ae5-11eb-8df0-2636a63b24ed.png)

![image](https://user-images.githubusercontent.com/40306834/101768244-9fed8f80-3ae5-11eb-873d-a18c6824163a.png)

</details>

<details>
  <summary>Dark mode</summary>

![image](https://user-images.githubusercontent.com/40306834/101768590-0d012500-3ae6-11eb-9343-e036bef4f6c5.png)

On button hover:

![image](https://user-images.githubusercontent.com/40306834/101768622-1c806e00-3ae6-11eb-9ab1-4cd82e5f6483.png)


</details>

michalsmiarowski commented:
For the color inversion the code was written based on a #338 Pull Request (WIP Dark Mode). It will require similar changes to inverse a colors in darkMode (changin $black/$white to var(--body-foreground-color)/var(--body-background-color)) + swapping --svg-main-color i --svg-secondary-color variables when in dark Mode (same as swapping foreground and background color in #338.

r-czajkowski reviewed (changes requested):
Left a couple of notes. Looks good 👍

r-czajkowski commented on src/css/home.scss:
```diff
@@ -242,6 +242,109 @@
+
+        .exchange-list-container {
+            .section-title {
+                font-size: 3.04rem;
```
I just wonder if these breakpoints are necessary ... 🤔 The other `section title` on the page don't have breakpoints and we already handle breakpoints for the `section-title` in `typography.scss`

r-czajkowski commented on src/components/lib/ExchangeList.js:
```diff
@@ -0,0 +1,73 @@
+import BalancerLogo from "../svgs/BalancerLogo"
+import classNames from "classnames"
+
+const getExchangeListItems = () => {
```
Another thought, we are in Gatsby and Netlify cms world. Maybe we should display data in a similar way to the [integration section](https://github.com/keep-network/tbtc-website/blob/master/src/components/lib/Integrations.js) and handle it with cms plugin?

Shadowfiend replied:
^^ this is definitely the ideal scenario, though this is a great intermediate state/building block.

r-czajkowski commented on src/css/home.scss:
```diff
@@ -242,6 +242,109 @@
+        margin-bottom: 3.12rem;
+        background-color: #{$black};
+
+        .exchange-list-container {
```
Maybe let's add some `margin-bottom` to the `exchange-list-container`. Thoughts?

r-czajkowski commented:
Could you please add screenshots to the PR description? Also, please let's link an issue with `Closes: #335` in PR description.

Shadowfiend commented on src/components/svgs/balancer.svg:
```diff
@@ -0,0 +1,6 @@
+<svg width="30" height="24" viewBox="0 0 30 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
For CMS purposes I think we should probably move away from SVGs anyway.

pdyraga commented:
We should also add Kraken to this list cc @liz-shinn

liz-shinn commented:
>We should also add Kraken to this list cc @liz-shinn

@Gluzman Looks like there's another exchange needed here for the tbtc website

michalsmiarowski commented:
This PR is ready for another review. I fixed the issues you guys mentioned and added some more changes. Please see the first post of this PR to view the full list of changes with examples.
